app/controllers/administrative-units/administrative-unit/governing-bodies/governing-body/mandatory/new.js

```diff
@@ -10,9 +10,11 @@ export default class AdministrativeUnitsAdministrativeUnitGoverningBodiesGoverni
   @service router;
   @service store;
 
-  queryParams = ['personId'];
+  queryParams = ['personId', 'positionId'];
```

Comment:
This isn't enough to make the position load properly (see the errors found in the global comment). Now it loads when `handleMandateRoleSelect` is called, so when an actual change happens through the power select of `MandateRoleSelect`. But here, if it's provided by a query param, we want it to load directly without having to do some clicks to change it in the power select.

We could for example add a constructor that would check if the query param is provided, and if so load the position.

This also means that as you load the position here, you can just pass it down to the `MandateRoleSelect` component as `@selected` and remove the loading logic from there (`loadRecord`)

Reply:
indeed, I ended up doing the initialization in the route